File: Resueltos_por_competidor/Gabriella_Castro/CowsAndCars.java
/*
 * Autor: Gabriella Castro
 * Problema: Cows and Cars
 * Juez online: VJudge
 * Veredicto: Accepted
 * URL: https://vjudge.net/problem/UVA-10491
 */

import java.util.Scanner;

public class Main{
	
	public static void main(String[] args) {
		
		Scanner
		sc = new Scanner(System.in);
		
		while(sc.hasNextInt()) {
			
			int vacas = sc.nextInt();
			int carros = sc.nextInt();
			int puertasAb = sc.nextInt();
			
			
			double arriba = carros * (vacas + carros - 1);
			
			double abajo = (carros + vacas) * (vacas + carros - puertasAb - 1);
			
			double rest = arriba / abajo;
			
			System.out.printf("%.5f",rest);
			System.out.println();
			
		}
		
	}
	
}

File: Resueltos_por_competidor/Tomás_Triana_Galvis/CowsAndCars.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;
/*
 * Autor: Tomás Triana Galvis
 * Problema: Cows and Cars
 * Juez online: UVA 10491
 * Veredicto: Accepted
 * Url: https://onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=1432
 **/ 
int main()
{
    double a,b,c,x,y,ans;
    while(cin>>a>>b>>c){
        x= a+b;
        y=(a+b)-(c+1);
        ans =(a/x)*(b/y)+((b-1)/y)*(b/x);
        printf("%.5f\n",ans);
    }



}
